### PR TITLE
build: bump FOCA from v0.6.0 to v0.10.0

### LINKED ISSUE #21: Update FOCA Version
Update the cloud registry to the latest FOCA version.

### PR DESCRIPTION
**Pull request by kushagra189**
## Description

This PR includes the changes required for migrating `cloud-registry` from `FOCA` `v0.6.0` -> `v0.10.0`. Changes include : 

* implement required changes imposed by FOCA API changes
* conversion of `endpoints` config to custom config and corresponding model definition creation
* update documentation & tests accordingly
* bump app version from `v0.1.0` to `v0.2.0`

Fixes #21 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist:

- [x] My code follows the [style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/contributing_guidelines.md#language-specific-guidelines) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage
- [x] I have added docstrings following the [Python style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/python.md) of this project to all new modules, classes, methods and functions are documented with docstrings following; I have updated any previously existing docstrings, if applicable
- [x] I have updated any sections of the app's documentation that are affected by the proposed changes, if applicable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ available [here][docs-api].
 Once [deployed](#Installation), the API will be served here:
 
 ```text
-http://localhost:8080/ga4gh/registry/v1/
+http://localhost:8080/ga4gh/registry/v1/ui/
```
Comment (uniqueg) [inline README.md]: That's not true. The _API_ itself will indeed be hosted at `http://localhost:8080/ga4gh/registry/v1/`, the _Swagger UI_ will be hosted at `http://localhost:8080/ga4gh/registry/v1/ui/`

File: cloud_registry/config.yaml [hunk 2]
```diff
@@ -93,36 +93,34 @@ exceptions:
...
+            id: "ELIXIR_CLOUD_SERVICE_REGISTRY_1"
+            name: "ELIXIR_CLOUD"
+            type:
+                group: "org.ga4gh"
+                artifact: "service-registry"
+                version: "1.0.0"
+            description: "Service registry for the ELIXIR Cloud network."
+            organization:
+                name: "ELIXIR Cloud & AAI"
+                url: "https://github.com/elixir-cloud-aai/elixir-cloud-aai"
```
Comment (uniqueg) [inline cloud_registry/config.yaml]: I guess it's time that we start referencing our new landing page here: https://elixir-cloud.dcc.sib.swiss/
You can put the same for the `contactUrl`

File: cloud_registry/config.yaml [hunk 3]
```diff
@@ -93,36 +93,34 @@ exceptions:
...
+            contactUrl: "https://github.com/elixir-cloud-aai/elixir-cloud-aai"
+            documentationUrl: "https://github.com/elixir-cloud-aai/elixir-cloud-aai"
```
Comment (uniqueg) [inline cloud_registry/config.yaml]: I guess documentation would be available at https://github.com/elixir-cloud-aai/cloud-registry for now

File: cloud_registry/config.yaml [hunk 4]
```diff
@@ -93,36 +93,34 @@ exceptions:
...
+            createdAt: '2020-11-04T12:58:19Z'
+            updatedAt: '2020-11-04T12:58:19Z'
```
Comment (uniqueg) [inline cloud_registry/config.yaml]: Guess you could update that as well to today's date. Of course, this would ideally be populated automatically, so perhaps open an issue for that (but not very high priority, certainly we shouldn't address it during the hackathon)

File: cloud_registry/config.yaml [hunk 5]
```diff
@@ -93,36 +93,34 @@ exceptions:
...
+            environment: "dev"
+            version: "1.0.0-dev-201109"
```
Comment (uniqueg) [inline cloud_registry/config.yaml]: Not quite sure where this is from, but I don't think that's accurate. We should probably put the version of the service. Again, best to populate this automatically. And again, let's have an issue for that but don't take care of it this week

File: cloud_registry/ga4gh/registry/service_info.py [hunk 6]
```diff
@@ -31,14 +31,14 @@ def __init__(self) -> None:
...
-        self.host_name = conf['service']['external_host']
-        self.external_port = conf['service']['external_port']
-        self.api_path = conf['service']['api_path']
-        self.conf_info = conf['service_info']
+        conf = current_app.config.foca.custom.endpoints
+        self.url_prefix = conf.service.url_prefix
+        self.host_name = conf.service.external_host
+        self.external_port = conf.service.external_port
+        self.api_path = conf.service.api_path
+        self.conf_info = conf.service_info
```
Comment (uniqueg) [inline cloud_registry/ga4gh/registry/service_info.py]: You may want to put the `.dict()` already here if we don't need to use `self.conf_info` as a Pydantic model anywhere in the code (and we probably don't). So basically, you can revert the change in line 79

File: cloud_registry/config.yaml (hunk 5 above)
Comment (uniqueg) [reply cloud_registry/config.yaml]: Or is this the version of the API? Either way, we store the API file somewhere, so we should be able to extract that info and populate automatically - in another, low priority issue

File: cloud_registry/service_models/custom_config.py [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+        pydantic.ValidationError: The class was instantianted with an illegal
+            data type.
+
+    Example:
+        >>> OrganizationConfig()
+        OrganizationConfig(name='ELIXIR Cloud & AAI', url='https://github.com/\
+elixir-cloud-aai/elixir-cloud-aai')
+    """
+    name: str = "ELIXIR Cloud & AAI"
+    url: str = "https://github.com/elixir-cloud-aai/elixir-cloud-aai"
```
Comment (uniqueg) [inline cloud_registry/service_models/custom_config.py]: We may want to set these as required, i.e., removing the defaults. Else we run the risk that other orgs deploy this service and end up referencing _our_ org.

File: cloud_registry/service_models/custom_config.py [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+    name: str = "ELIXIR_CLOUD"
+    type: TypeConfig = TypeConfig()
+    description: str = "Service registry for the ELIXIR Cloud network."
+    organization: OrganizationConfig = OrganizationConfig()
+    contactUrl: str = "https://github.com/elixir-cloud-aai/elixir-cloud-aai"
+    documentationUrl: str = "https://github.com/elixir-cloud-aai/elixir-cloud-aai"
+    createdAt: str = '2020-11-04T12:58:19Z'
+    updatedAt: str = '2020-11-04T12:58:19Z'
+    environment: str = "dev"
+    version: str = "1.0.0-dev-201109"
```
Comment (uniqueg) [inline cloud_registry/service_models/custom_config.py]: Similarly, we may want to remove (most of) the defaults here and require users to provide these (see comment above). Ideally, these requirements should be documented in the `README.md`. The examples in both cases would need to be adapted accordingly, as well as in the models that depend on these - you can pass some short dummy contents.

**Review by uniqueg — CHANGES_REQUESTED** [review]
Thanks :) Some minor issues. Another one is that perhaps you could bump the `cloud-registry` version as well, as we should probably tag this for a new release before adding access control

File: README.md (hunk 1 above)
Comment (kushagra189) [reply README.md]: Ah, my bad.

File: cloud_registry/config.yaml (hunk 2 above)
Comment (kushagra189) [reply cloud_registry/config.yaml]: done

File: cloud_registry/config.yaml (hunk 3 above)
Comment (kushagra189) [reply cloud_registry/config.yaml]: done

File: cloud_registry/config.yaml (hunk 4 above)
Comment (kushagra189) [reply cloud_registry/config.yaml]: done, added issue.

File: cloud_registry/config.yaml (hunk 5 above)
Comment (kushagra189) [reply cloud_registry/config.yaml]: Not sure what it is. Please update?

File: cloud_registry/ga4gh/registry/service_info.py (hunk 6 above)
Comment (kushagra189) [reply cloud_registry/ga4gh/registry/service_info.py]: done

**Review by uniqueg — CHANGES_REQUESTED** [review]
Awesome! Will approve when the versioning stuff is in

File: cloud_registry/__init__.py [hunk 9]
```diff
@@ -1 +0,0 @@
-__version__ = '0.1.0'
```
Comment (uniqueg) [inline cloud_registry/__init__.py:1]: Sorry to nag, but can you include a module-level docstring, like the one here?
https://github.com/elixir-cloud-aai/proTES/blob/dev/pro_tes/__init__.py

So basically,

```py
"""Cloud Registry app package root."""
```

**Review by uniqueg — CHANGES_REQUESTED** [review]
Sorry...

**Review by uniqueg — APPROVED** [review]
